packages/@aws-cdk/aws-appmesh/lib/shared-interfaces.ts

```diff
@@ -167,6 +169,48 @@ class FileAccessLog extends AccessLog {
   }
 }
 
+/**
+ * Represents the properties needed to define TLS validation context
+ */
+export interface TlsValidationContext {
+  /**
+   * Reference to where to retrieve the trust chain.
+   */
+  readonly trust: TlsValidationContextTrust;
+}
+
+/**
+ * Represents the properties needed to define client policy
+ */
+export interface ClientPolicy {
```

Comment:
The way this interface is structured right now, it is more of a `TlsClientPolicy` and not `ClientPolicy`. 

If we have `enforce`, `ports`, etc. at the top level client policy, we cannot extend the top level `ClientPolicy` cleanly. Would `enforce` apply to `TLS Validation` or to some other client policy? It's not clear.


We could just rename this class to `TlsClientPolicy` and leave it as a top level construct on our constructs. For example, backends would have a `TlsClientPolicy` object. Alternatively we could have:

```typescript
interface ClientPolicy {
   readonly tlsClientPolicy?: TlsClientPolicy;
}
```

Which would still group all the client policies together as we add more in the future. May be good to see what the best approach here is from Adam from a CDK perspective

Reply:
I was thinking about this too and I like the option for renaming the current `ClientPolicy` to `TlsClientPolicy` to start with.

Creating the top-level `ClientPolicy` might be more future-proof but I feel like this structuring is what the CDK design guideline is asking to avoid.